### ISSUE TITLE
How to stream multiply IP-Cameras

### ISSUE THREAD

**Issue by baur** [k-yle/rtsp-relay#1]
Hi, good work!!!

Works fine for one camera, but how to stream multiple IP-Cameras?

In my case, every client requests different cameras and they can do it simultaneously (> 500 IP cameras). 
Is there some way to create a dynamic port if was requested another camera, such this [one](https://github.com/databilim/rtsp-stream/blob/master/app.js)

and  I'd like to close every stream after 5 min after starting in order to save server resources.

thanks!

**Comment by baur** [comment]
[here](https://github.com/Zveroslav/node-rtsp-stream-jsmpeg/issues/1) was proposed to use Docker but it's too difficult for me, especially since I have a windows server.

**Comment by k-yle** [comment]
Hi, I'm still actively developing this module, but for now something like this will work:

I will be making changes over the next few days so that this is much more expandible and performant.  

```js
const express = require('express');
const app = express();

const { proxy, killAll } = require('rtsp-relay')(app);

app.ws('/api/stream/1', proxy({
  url: `rtsp://admin:admin@10.0.1.1:554/feed`
}));

app.ws('/api/stream/2', proxy({
  url: `rtsp://admin:admin@10.0.1.2:554/feed`
}));

// close all connections every 5 minutes 
setInterval(killAll, 5 * 60 * 1000);

app.listen(2000);
```

**Comment by baur** [comment]
Thanks a lot for answer!!!
but rendered same camera, though was different IP addresses

```
app.get('/', (req, res) =>
  res.send(`
  <canvas id='canvas1'></canvas>
  <canvas id='canvas2'></canvas>

  <script src='jsmpeg.min.js'></script>
  <script>
    new JSMpeg.Player('ws://' + location.host + '/api/stream/1', {
      canvas: document.getElementById('canvas1')
    })

    new JSMpeg.Player('ws://' + location.host + '/api/stream/2', {
      canvas: document.getElementById('canvas2')
    })    
  </script>
`),
);
```

**Comment by k-yle** [comment]
It's not an issue with the port - this module only uses one port for HTTP and Websocket traffic. I'll take a look into it later this week.

**Comment by hongson23** [comment]
hello @k-yle 
Thank you for your code

It works perfectly, i would like to play multiple camera, so do you have solution?

thanks

**Comment by henrik-muehe** [comment]
Move this line

```
  const Inbound = new InboundStreamWrapper();
```

inside the proxy() method and this works. It seems to be explicitly built to stop multiple streams in parallel however there's no clear reason stated as to why. Either way, moving that line works well for me.

**Comment by k-yle** [comment]
I'm happy to accept PRs. won't have time to work on this for a few weeks

Sent from Outlook for Android

________________________________

**Comment by k-yle** [comment]
thanks @henrik-muehe I've fixed and released this in v1.1.0

**Comment by hongson23** [comment]
it works, thank you so much!
